src/libdredd/src/mutate_visitor.cc

```diff
@@ -77,10 +84,46 @@ bool MutateVisitor::TraverseDecl(clang::Decl* decl) {
   return true;
 }
 
+bool MutateVisitor::VisitUnaryOperator(clang::UnaryOperator* unary_operator) {
+  if (NotInFunction()) {
+    // Only consider mutating unary expressions that occur inside functions.
+    return true;
+  }
+
+  // Check that the unary expression and its arguments have source
+  // ranges that are part of the main file. In particular, this avoids
+  // mutating expressions that directly involve the use of macros (though it
+  // is OK if sub-expressions of arguments use macros).
+  if (GetSourceRangeInMainFile(compiler_instance_.getPreprocessor(),
+                               *unary_operator)
+          .isInvalid() ||
+      GetSourceRangeInMainFile(compiler_instance_.getPreprocessor(),
+                               *unary_operator->getSubExpr())
+          .isInvalid()) {
+    return true;
+  }
+
+  // Don't mutate unary plus
```

Comment:
extend comment slightly to remark that this is because doing so is unlikely to lead to a mutation that differs from inserting a different unary operator.

Reply:
Added more explanation.